lib/parser.ts

```diff
@@ -0,0 +1,414 @@
+interface ParsedElement {
```

Comment:
I would do a zod definition for this then derive the type like this:

```ts
export const ParsedElementSchema = z.object({
  // ...
})
export type ParsedElement = z.infer<ParsedElementSchema>

```

This way you can also do a runtime check if need be- it's a good technique for parsers to make sure your content is always valid